adam/language_specific/english/english_language_generator.py

```diff
@@ -96,8 +100,25 @@ class _Generation:
         ] = dict()
 
         def generate(self) -> ImmutableSet[LinearizedDependencyTree]:
-            for _object in self.situation.objects:
-                self._translate_object_to_noun(_object)
+            # For now, only apply quantifiers to object-only situations
+            if self.situation.actions == ImmutableSet.empty():
+                # Get number of objects of each type
+                node_counts: Dict[OntologyNode, int] = dict()
```

Comment:
Try `Counter` here: https://docs.python.org/2/library/collections.html#collections.Counter

Reply:
I totally forgot this existed...

Reply:
Done